mlx/backend/metal/ternary.cpp

```diff
@@ -0,0 +1,136 @@
+// Copyright © 2024 Apple Inc.
+
+#include "mlx/backend/common/ternary.h"
+#include "mlx/backend/metal/device.h"
+#include "mlx/backend/metal/utils.h"
+#include "mlx/primitives.h"
+
+#ifndef MLX_METAL_JIT
```

Comment:
This is a minor downside of going the constexpr route. But the only place we really need to use the preprocessor.